gr-fec: removed inconsistent header and implementation for RS code

StudHamza opened the pull request:
<!--- The title of the PR should summarize the change implemented. -->
<!--- Example commit message format: -->
<!--- `module: summary of change` -->
<!--- (leave blank) -->
<!--- `details of what/why/how an issue was addressed` -->
<!--- Keep subject lines to 50 characters (but 72 is a hard limit!) -->
<!--- characters. Refer to the [Revision Control Guidelines](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md#revision-control-guidelines) section of the coding guidelines -->

## Description
<!--- Provide a general summary of your changes in the title above -->
Implemented solution 1 that was proposed by @daniestevez:
Remove the `_int` function definitions from `rs.h` and the removed fec/int.h.



## Related Issue
<!--- Refer to any related issues here -->
Fixes #6985 
<!--- as this will allow Github to automatically close the related Issue -->

## Which blocks/areas does this affect?
<!--- Include blocks that are affected and some details on what -->
<!--- areas these changes affect, such as performance. -->

## Testing Done
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you -->
<!--- ran to see how your change affects other areas of the code, -->
<!--- etc. Then, include justifications for how your tests -->
<!--- demonstrate those affects. -->
No errors during build and install.

## Checklist
<!--- Go over all the following points, and put an `x` in all the
<!--- boxes that apply. Note that some of these may not be valid -->
<!--- for all PRs. -->

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [x] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

StudHamza commented:
forgot to sign my commit, but now I signed it and pushed to main

marcusmueller commented on gr-fec/lib/reed-solomon/rstest.c:
```diff
@@ -95,20 +94,8 @@ int main()
             }
             errs = exercise_char(handle, Tab[i].ntrials);
         } else {
-#ifdef ALL_VERSIONS
-            if ((handle = init_rs_int(Tab[i].symsize,
-                                      Tab[i].genpoly,
-                                      Tab[i].fcs,
-                                      Tab[i].prim,
-                                      Tab[i].nroots)) == NULL) {
-                printf("init_rs_int failed!\n");
-                continue;
-            }
-            errs = exercise_int(handle, Tab[i].ntrials);
-#else
             printf("init_rs_init support is not enabled\n");
             exit(1);
-#endif
```
Is this proper or do we need to 
```c++
} else {
#if !defined(ALL_VERSIONS)
printf(…);
exit(1);
#endif
}
```

?

StudHamza replied:
You are right, I will fix it then rebuild.

daniestevez replied:
Now that we're removing the `_int` functions, we can also remove the test cases in `Tab[]` that have `symsize > 8`, since those are exclusively for the `_int` functions, and also get rid of the `if (Tab[i].symsize <= 8)` and all the `else`.

StudHamza replied:
what if the the is not `Tab[i].symsize <= 8`, then it will not implement the code in the else statement `else{
#if !defined(ALL_VERSIONS)
            printf("init_rs_init support is not enabled\n");
            exit(1);
#endif
        }`
      I dont think we should remove the `if (Tab[i].symsize <= 8)` and all the `else`.

daniestevez replied:
Just to be clear, my suggestion was as follows:

Amend
```c++
struct {
    int symsize;
    int genpoly;
    int fcs;
    int prim;
    int nroots;
    int ntrials;
} Tab[] = {
    { 2, 0x7, 1, 1, 1, 10 },
    { 3, 0xb, 1, 1, 2, 10 },
    { 4, 0x13, 1, 1, 4, 10 },
    { 5, 0x25, 1, 1, 6, 10 },
    { 6, 0x43, 1, 1, 8, 10 },
    { 7, 0x89, 1, 1, 10, 10 },
    { 8, 0x11d, 1, 1, 32, 10 },
    { 8, 0x187, 112, 11, 32, 10 }, /* Duplicates CCSDS codec */
#ifdef ALL_VESIONS
    { 9, 0x211, 1, 1, 32, 10 },
    { 10, 0x409, 1, 1, 32, 10 },
    { 11, 0x805, 1, 1, 32, 10 },
    { 12, 0x1053, 1, 1, 32, 5 },
    { 13, 0x201b, 1, 1, 32, 2 },
    { 14, 0x4443, 1, 1, 32, 1 },
    { 15, 0x8003, 1, 1, 32, 1 },
    { 16, 0x1100b, 1, 1, 32, 1 },
#endif
    { 0, 0, 0, 0, 0 },
};
```
to
```c++
struct {
    int symsize;
    int genpoly;
    int fcs;
    int prim;
    int nroots;
    int ntrials;
} Tab[] = {
    { 2, 0x7, 1, 1, 1, 10 },
    { 3, 0xb, 1, 1, 2, 10 },
    { 4, 0x13, 1, 1, 4, 10 },
    { 5, 0x25, 1, 1, 6, 10 },
    { 6, 0x43, 1, 1, 8, 10 },
    { 7, 0x89, 1, 1, 10, 10 },
    { 8, 0x11d, 1, 1, 32, 10 },
    { 8, 0x187, 112, 11, 32, 10 }, /* Duplicates CCSDS codec */
    { 0, 0, 0, 0, 0 },
};
```

Then remove the condition
```c++
if (Tab[i].symsize <= 8)
```
since now this will always be true, taking the code inside the `if` outside, so that it always runs. Finally, remove
```c++
        } else{
#if !defined(ALL_VERSIONS)
            printf("init_rs_init support is not enabled\n");
            exit(1);
#endif
        }
```

By the way, this also makes me realize that we aren't building the test with `-DALL_VERSIONS`, and maybe we should.

StudHamza commented:
Building is successful, removed references of `_int` function in `gr-fec/lib/reed-solomon/rs.3` (a man page) and gr-`fec/include/gnuradio/fec/rs.h`, and `gr-fec/lib/reed-solomon/rstest.c`

daniestevez commented:
@StudHamza there are changes on a QTGUI file that you probably didn't mean to include in the PR.

daniestevez commented:
There is also code related to the `_int` functions in `exercise.c`. Probably this should be removed too (even though I think it's not being built), since we've also removed it from `rstest.c`.

StudHamza commented:
I re-based and forced push, plus removed `_int` functions in `exercise.c`

daniestevez commented on gr-fec/lib/reed-solomon/exercise.c:
```diff
@@ -22,8 +22,6 @@
 #include "fixed.h"
 #define EXERCISE exercise_ccsds
 #elif defined(BIGSYM)
```
Now that we've removed the `_int` functions, we don't need `BIGSYM` (this part of the `#elif` was now empty, and probably wouldn't build with `-DBIGSYM`).
```suggestion
```

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```diff
@@ -62,12 +51,12 @@ and \fBdecode_rs_8\fR.
 arguments are as follows:
 
 \fBsymsize\fR gives the symbol size in bits, up to 8 for \fBinit_rs_char\fR
-or 32 for \fBinit_rs_int\fR on a machine with 32-bit ints (though such a
+or 32 for on a machine with 32-bit ints (though such a
 huge code would exhaust memory limits on a 32-bit machine). The resulting
```
This sentence is not correct now. Once we have removed the `_int` functions, the symbol size can only be 8 bits.

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```diff
@@ -90,16 +79,15 @@ per code block.
 into the \fBparity\fR array, right justified.
 
-The \fBdecode_rs_char\fR and \fBdecode_rs_int\fR functions correct
+The \fBdecode_rs_char\fR and functions correct
```
```suggestion
The \fBdecode_rs_char\fR functions corrects
```

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```suggestion
The \fBfree_rs_char\fR function frees the internal
```

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```suggestion
space allocated by the \fBinit_rs_char\fR function.
```

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```diff
@@ -137,7 +125,7 @@ and using the resulting pointer with \fBencode_rs_char\fR and
 
 .SH RETURN VALUES
-\fBinit_rs_int\fR and \fBinit_rs_char\fR return a pointer to an internal
+\fBinit_rs_char\fR return a pointer to an internal
```
```suggestion
\fBinit_rs_char\fR returns a pointer to an internal
```

StudHamza commented:
Fixed grammar errors in man page `rs.3`, removed `if (Tab[i].symsize <= 8)` and `#ifdef ALL_VESIONS
    { 9, 0x211, 1, 1, 32, 10 },
    { 10, 0x409, 1, 1, 32, 10 },
    { 11, 0x805, 1, 1, 32, 10 },
    { 12, 0x1053, 1, 1, 32, 5 },
    { 13, 0x201b, 1, 1, 32, 2 },
    { 14, 0x4443, 1, 1, 32, 1 },
    { 15, 0x8003, 1, 1, 32, 1 },
    { 16, 0x1100b, 1, 1, 32, 1 },
#endif`

daniestevez commented on gr-fec/lib/reed-solomon/rs.3:
```diff
@@ -62,13 +51,11 @@ and \fBdecode_rs_8\fR.
 To use the general purpose RS encoder or decoder (i.e.,
-the _char or _int versions), the user must first
-call \fBinit_rs_int\fR or \fBinit_rs_char\fR as appropriate. The
+the _char ), the user must first
```
```suggestion
the _char version), the user must first
```

daniestevez commented on gr-fec/lib/reed-solomon/rstest.c:
```diff
@@ -76,40 +65,27 @@ int main()
-            errs = exercise_int(handle, Tab[i].ntrials);
-#else
+        } 
+#if !defined(ALL_VERSIONS)
             printf("init_rs_init support is not enabled\n");
             exit(1);
 #endif
```
What is the intent here?

daniestevez replied:
This piece of code wasn't really doing anything in the original code. I imagine that the intent was that if by accident a test case with `symsize > 8` was declared outside the `#ifdef ALL_VERSIONS`, then the test would print and error and finish, because `init_rs_int()`, which is necessary to test that test case, was only used when `ALL_VERSIONS` was defined. Now that we've removed the `_int()` functions, this piece of code is not only not doing anything, but also it doesn't play any role anymore, since now there should be no test cases with `symsize > 8` defined at all.

By the way, I realized that the original code uses `#ifdef ALL_VESIONS` in one place instead of `#ifdef ALL_VERSIONS`, but this doesn't matter because we've now deleted all this.

daniestevez reviewed (approved):
All looks good to me now.
